### PR TITLE
spades: Update to 4.0.0 and add linux-aarch64 build

### PR DESCRIPTION
**Pull request by martin-g**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by martin-g** [comment]
The Linux ARM64 build will have to wait for the `next` release of SPAdes: https://github.com/ablab/spades/issues/1062#issuecomment-2019491378

File: recipes/spades/meta.yaml [hunk 1]
```diff
@@ -21,28 +21,27 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - llvm-openmp  # [osx]
-    - openmp  # [linux]
+    - libgomp  # [linux]
     - cmake
     - make
     - pkg-config
-    - sysroot_linux-64 =2.17 # [linux]
+    - sysroot_linux-64 >=2.17 # [linux]
```
Comment (martin-g) [inline recipes/spades/meta.yaml:28]:
```suggestion
    - sysroot_linux-64 =2.17 # [linux]
```

File: recipes/spades/meta.yaml [hunk 2]
```diff
@@ -21,28 +21,27 @@ requirements:
...
   host:
     - llvm-openmp  # [osx]
-    - openmp  # [linux]
+    - libgomp  # [linux]
     - zlib
     - bzip2
-    - sysroot_linux-64 =2.17 # [linux]
+    - sysroot_linux-64 >=2.17 # [linux]
```
Comment (martin-g) [inline recipes/spades/meta.yaml:34]:
```suggestion
    - sysroot_linux-64 =2.17 # [linux]
```

File: recipes/spades/meta.yaml [hunk 3]
```diff
@@ -21,28 +21,27 @@ requirements:
...
   run:
     - llvm-openmp  # [osx]
-    - openmp  # [linux]
+    - libgomp  # [linux]
     - python >3.9
-    - sysroot_linux-64 =2.17 # [linux]
+    - sysroot_linux-64 >=2.17 # [linux]
```
Comment (martin-g) [inline recipes/spades/meta.yaml:39]:
```suggestion
    - sysroot_linux-64 =2.17 # [linux]
```

**Comment by martin-g** [comment]
```
mgrigorov in 🌐 euler-arm-22 in /tmp/spades took 44s 
❯ tar xvf spades-4.0.0-h8729679_0.tar.bz2 
info/run_exports.json
info/recipe/3.9.1/build.sh
info/recipe/3.12.0/build.sh
info/recipe/build.sh
info/hash_input.json
info/test/run_test.sh
info/index.json
info/recipe/0001-change-verbosity-remove-expr.patch
info/recipe/3.9.1/meta.yaml
info/recipe/post-link.sh
info/recipe/3.12.0/meta.yaml
info/recipe/meta.yaml.template
info/licenses/LICENSE
info/recipe/meta.yaml
info/files
info/recipe/conda_build_config.yaml
info/about.json
info/paths.json
info/git
bin/.spades-post-link.sh
share/spades/README.md
share/spades/sewage/usher_barcodes.csv
share/spades/configs/debruijn/toy.info
share/spades/configs/debruijn/isolate_mode.info
share/spades/configs/debruijn/metaplasmid_mode.info
share/spades/configs/debruijn/hmm_mode.info
share/spades/configs/debruijn/large_genome_mode.info
share/spades/configs/corrector/corrector.info
share/spades/configs/debruijn/plasmid_mode.info
share/spades/configs/debruijn/careful_mda_mode.info
share/spades/configs/debruijn/careful_mode.info
share/spades/configs/debruijn/rnaviral_mode.info
share/spades/configs/debruijn/construction.info
share/spades/configs/debruijn/detail_info_printer.info
share/spades/configs/debruijn/metaviral_mode.info
share/spades/configs/debruijn/sewage_mode.info
share/spades/configs/hammer/config.info
share/spades/configs/debruijn/distance_estimation.info
share/spades/configs/debruijn/mda_mode.info
share/spades/configs/debruijn/pe_params.info
share/spades/configs/debruijn/meta_mode.info
share/spades/configs/debruijn/config.info
share/spades/configs/debruijn/rna_mode.info
share/spades/configs/debruijn/simplification.info
share/spades/test_dataset/operons_1K.txt
share/spades/test_dataset/genes_1K.txt
share/spades/GPLv2.txt
bin/metaviralspades.py
bin/metaspades.py
bin/rnaspades.py
bin/plasmidspades.py
bin/metaplasmidspades.py
bin/coronaspades.py
bin/rnaviralspades.py
share/spades/joblib3/_compat.py
share/spades/spades_pipeline/common/__init__.py
share/spades/spades_pipeline/stages/__init__.py
share/spades/joblib3/testing.py
share/spades/spades_pipeline/executors/executors.py
share/spades/spades_pipeline/scripts/copy_files.py
share/spades/spades_pipeline/stages/stage.py
share/spades/spades_pipeline/executors/executor_save_yaml.py
share/spades/pyyaml3/loader.py
share/spades/joblib3/_multiprocessing_helpers.py
share/spades/spades_pipeline/stages/terminating_stage.py
share/spades/spades_pipeline/scripts/plasmid_glue.py
share/spades/pyyaml3/nodes.py
share/spades/spades_pipeline/stages/pipeline.py
share/spades/spades_pipeline/stages/breaking_scaffolds_stage.py
bin/spades_init.py
share/spades/spades_pipeline/stages/before_start_stage.py
share/spades/spades_pipeline/stages/check_test_stage.py
share/spades/spades_pipeline/stages/scaffold_correction_stage.py
share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py
share/spades/spades_pipeline/commands_parser.py
share/spades/spades_pipeline/common/parallel_launcher.py
share/spades/pyyaml3/events.py
share/spades/spades_pipeline/scripts/preprocess_contigs.py
share/spades/pyyaml3/error.py
share/spades/pyyaml3/tokens.py
share/spades/pyyaml3/dumper.py
share/spades/spades_pipeline/common/alignment.py
share/spades/spades_pipeline/executors/executor_local.py
share/spades/joblib3/my_exceptions.py
share/spades/joblib3/disk.py
share/spades/pyyaml3/cyaml.py
share/spades/spades_pipeline/scripts/correction_iteration_script.py
share/spades/spades_pipeline/scripts/check_test_script.py
share/spades/joblib3/_memory_helpers.py
share/spades/spades_pipeline/options_storage.py
share/spades/pyyaml3/serializer.py
share/spades/spades_pipeline/scripts/preprocess_interlaced_reads.py
share/spades/spades_pipeline/process_cfg.py
share/spades/joblib3/__init__.py
share/spades/spades_pipeline/scripts/compress_all.py
share/spades/pyyaml3/composer.py
share/spades/joblib3/logger.py
share/spades/spades_pipeline/common/SeqIO.py
share/spades/spades_pipeline/stages/correction_stage.py
share/spades/pyyaml3/reader.py
share/spades/spades_pipeline/stages/spades_iteration_stage.py
share/spades/pyyaml3/resolver.py
share/spades/joblib3/hashing.py
share/spades/spades_pipeline/common/sam_parser.py
share/spades/spades_pipeline/stages/error_correction_stage.py
share/spades/pyyaml3/__init__.py
share/spades/joblib3/func_inspect.py
share/spades/spades_pipeline/stages/preprocess_reads_stage.py
share/spades/pyyaml3/representer.py
share/spades/joblib3/format_stack.py
share/spades/joblib3/numpy_pickle.py
share/spades/spades_pipeline/stages/spades_stage.py
share/spades/joblib3/pool.py
share/spades/pyyaml3/parser.py
share/spades/pyyaml3/constructor.py
share/spades/spades_pipeline/support.py
bin/spades.py
share/spades/joblib3/parallel.py
share/spades/joblib3/memory.py
share/spades/pyyaml3/emitter.py
share/spades/pyyaml3/scanner.py
share/spades/spades_pipeline/options_parser.py
share/spades/test_dataset/operons_1K.gff
share/spades/test_dataset/genes_1K.gff
share/spades/VERSION
share/spades/LICENSE
share/spades/test_dataset/reference_1K.fa.gz
share/spades/biosynthetic_spades_hmms/KR.hmm.gz
share/spades/biosynthetic_spades_hmms/TE.hmm.gz
share/spades/biosynthetic_spades_hmms/CStart.hmm.gz
share/spades/biosynthetic_spades_hmms/AT.hmm.gz
share/spades/biosynthetic_spades_hmms/KS.hmm.gz
share/spades/biosynthetic_spades_hmms/AMP.hmm.gz
share/spades/test_dataset_plasmid/pl1.fq.gz
share/spades/test_dataset_plasmid/pl2.fq.gz
share/spades/test_dataset/ecoli_1K_1.fq.gz
share/spades/test_dataset/ecoli_1K_2.fq.gz
share/spades/coronaspades_hmms/Phan-CoV.hmm.gz
share/spades/coronaspades_hmms/Pfam-A.SARS-CoV-2.hmm.gz
bin/spades-bwa
bin/spades-convert-bin-to-fasta
bin/spades-corrector-core
bin/spades-hammer
bin/spades-kmercount
bin/spades-kmer-estimating
bin/spades-ionhammer
fbin/spades-gfa-split
bin/spades-read-filter
ile bin/spades-gbuilder
.bin/spades-gsimplifier
/bin/spades-gmapper
bi	bin/spades-core
share/spades/configs/ionhammer/ionhammer.cfg

mgrigorov in 🌐 euler-arm-22 in /tmp/spades took 7s 
❯ file ./bin/*
./bin/coronaspades.py:             symbolic link to spades.py
./bin/metaplasmidspades.py:        symbolic link to spades.py
./bin/metaspades.py:               symbolic link to spades.py
./bin/metaviralspades.py:          symbolic link to spades.py
./bin/plasmidspades.py:            symbolic link to spades.py
./bin/rnaspades.py:                symbolic link to spades.py
./bin/rnaviralspades.py:           symbolic link to spades.py
./bin/spades-bwa:                  ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-convert-bin-to-fasta: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-core:                 ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-corrector-core:       ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-gbuilder:             ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-gfa-split:            ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-gmapper:              ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-gsimplifier:          ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-hammer:               ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-ionhammer:            ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-kmer-estimating:      ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-kmercount:            ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades-read-filter:          ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
./bin/spades.py:                   Python script, ASCII text executable
./bin/spades_init.py:              Python script, ASCII text executable

mgrigorov in 🌐 euler-arm-22 in /tmp/spades 
❯ ./bin/spades-bwa 

Program: bwa (alignment via Burrows-Wheeler transformation)
Version: 0.7.16a-r1185-dirty
Contact: Heng Li <lh3@sanger.ac.uk>

Usage:   bwa <command> [options]

Command: index         index sequences in the FASTA format
         mem           BWA-MEM algorithm
         fastmap       identify super-maximal exact matches
         pemerge       merge overlapping paired ends (EXPERIMENTAL)
         aln           gapped/ungapped alignment
         samse         generate alignment (single ended)
         sampe         generate alignment (paired ended)
         bwasw         BWA-SW for long queries

         shm           manage indices in shared memory
         fa2pac        convert FASTA to PAC format
         pac2bwt       generate BWT from PAC
         bwtupdate     update .bwt to the new format
         bwt2sa        generate SA from BWT and Occ

Note: To use BWA, you need to first index the genome with `bwa index'.
      There are three alignment algorithms in BWA: `mem', `bwasw', and
      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'
      first. Please `man ./bwa.1' for the manual.
```

LGTM!

**Comment by martin-g** [comment]
@BiocondaBot please add label

**Review by mencian — APPROVED** [review]

**Pull request merged**